src/Domain/CoordinateMaps/BulgedCube.cpp

```diff
@@ -35,29 +35,22 @@ class RootFunction {
         z_sq_(z_sq) {}
 
   double operator()(const double rho) const noexcept {
-    if (LIKELY(physical_r_squared_ != 0.0)) {
```

Comment:
To avoid potential future bugs, what do you think of adding an assertion in the `RootFunction` constructor to check that `physical_r_squared` is non-zero?

Additionally, I think it would be helpful to add a comment (to the constructor) to document the r > 0 assumption being made.

Reply:
I've added the ASSERT to the constructor. Is the error message enough to not need a C++ comment too?

Reply:
I agree, the error message is documentation enough in this case 👍